apps/meteor/app/lib/server/functions/saveUserIdentity.ts

```diff
@@ -119,14 +120,11 @@ async function updateUsernameReferences({
 			await fileStore.model.updateFileNameById(previousFile._id, username);
 		}
 
-		await Messages.updateAllUsernamesByUserId(user._id, username);
-		await Messages.updateUsernameOfEditByUserId(user._id, username);
-
-		const cursor = Messages.findByMention(previousUsername);
-		for await (const msg of cursor) {
-			const updatedMsg = msg.msg.replace(new RegExp(`@${previousUsername}`, 'ig'), `@${username}`);
-			await Messages.updateUsernameAndMessageOfMentionByIdAndOldUsername(msg._id, previousUsername, username, updatedMsg);
-		}
+		await Message.updateUserReferences({
+			username,
+			previousUsername,
+			userId: user._id,
+		});
```

Comment:
```suggestion
		await Message.updateUserReferences(username, previousUsername, user._id);
```
Changing this in order to follow the suggestion below.